### ISSUE TITLE
Properties of template reference variables cannot be used in the referenced element in Ivy built libs

### ISSUE THREAD

**Issue by darress** [angular/angular#35603]
# 🐞 bug report

### Affected Package
Probably @angular/compiler

### Is this a regression?
The first version I detected this problem in is 9.0.0.

### Description
If I try to use a property of a template reference variable in the referenced component itself, it throws a `error TS2531: Object is possibly 'null'`, no matter if I use !. or ?. or any other methods except for $any(), which works.
This only occurs when trying to build the library with Ivy.

Examples:
```typescript
      <lib-other #other1 [someInput]="other1!.someProp"></lib-other>
      <lib-other #other2 [someInput]="other2?.someProp"></lib-other>
      <lib-other #other3 [someInput]="other3 && other3.someProp"></lib-other>
```

## 🔬 Minimal Reproduction
https://github.com/darress/angular9_lib_bugs/tree/hashreference_null

Try to build with `ng build my-lib`, the compilation errors will show.

## 🔥 Exception or Error
```
projects/my-lib/src/lib/my-lib.component.ts:7:39 - error TS2531: Object is possibly 'null'.

7       <lib-other #other1 [someInput]="other1!.someProp"></lib-other>
                                        ~~~~~~~~~~~~~~~~
projects/my-lib/src/lib/my-lib.component.ts:8:39 - error TS2531: Object is possibly 'null'.

8       <lib-other #other2 [someInput]="other2?.someProp"></lib-other>
                                        ~~~~~~~~~~~~~~~~
projects/my-lib/src/lib/my-lib.component.ts:9:49 - error TS2531: Object is possibly 'null'.

9       <lib-other #other3 [someInput]="other3 && other3.someProp"></lib-other>
                                                  ~~~~~~~~~~~~~~~
```

## 🌍  Your Environment

**Angular Version:**
<pre><code>

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 9.0.3
Node: 10.14.2
OS: win32 x64

Angular: 9.0.2
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router
Ivy Workspace: Yes

Package                            Version
------------------------------------------------------------
@angular-devkit/architect          0.900.3
@angular-devkit/build-angular      0.900.3
@angular-devkit/build-ng-packagr   0.900.3
@angular-devkit/build-optimizer    0.900.3
@angular-devkit/build-webpack      0.900.3
@angular-devkit/core               9.0.3
@angular-devkit/schematics         9.0.3
@angular/cli                       9.0.3
@ngtools/webpack                   9.0.3
@schematics/angular                9.0.3
@schematics/update                 0.900.3
ng-packagr                         9.0.1
rxjs                               6.5.4
typescript                         3.7.5
webpack                            4.41.2
</code></pre>

**Anything else relevant?**
Nothing special.

**Comment by ericmartinezr** [comment]
Dup of https://github.com/angular/angular/issues/35372 and https://github.com/angular/angular/issues/35522 (workaround here)

**Comment by alxhub** [comment]
(closing as a duplicate, but this should be fixed soon by #35622)
